### PR TITLE
fix(ads): Children outside ADS location, flight endpoint fix

### PR DESCRIPTION
**Pull request by kksteini**
## What

Children, who were granted a flightcode based on their custodians location, but lived outside the ADS required area themselves would get denied on the `api/public/discounts/{discountCode}/flights` endpoint with 403.

Leveraging the cache from userService to fix that.

## Checklist:

- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] Formatting passes locally with my changes
- [X] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/air-discount-scheme/backend/src/app/modules/flight/flight.controller.ts [hunk 1]
```diff
@@ -213,9 +215,36 @@ export class PublicFlightController {
...
     }
 
-    const meetsADSRequirements = this.flightService.isADSPostalCode(
+    let meetsADSRequirements = this.flightService.isADSPostalCode(
       user.postalcode,
     )
+
+    // TODO: this is a quickly made temporary hotfix and should be rewritten
+    // along when the nationalregistry module is rewritten for the client V2 completely
+    if (!meetsADSRequirements) {
```
Comment (JuliusGG) [inline apps/air-discount-scheme/backend/src/app/modules/flight/flight.controller.ts:224]: You might consider adding a age check as well incase the edge case where user is actually adult and shouldn't be able to proceed.

**Review by kristofer-lp8f — APPROVED** [review]
Much nesting, but LGTM 👍

File: apps/air-discount-scheme/backend/src/app/modules/flight/flight.controller.ts (hunk 1 above)
Comment (kksteini) [reply apps/air-discount-scheme/backend/src/app/modules/flight/flight.controller.ts:224]: The age check already takes place for when the cache is set here:
https://github.com/island-is/island.is/blob/d50c699ab9237f809a8c8aa643f110c47e1d080e/apps/air-discount-scheme/backend/src/app/modules/user/user.service.ts#L95-L111

Comment (JuliusGG) [reply apps/air-discount-scheme/backend/src/app/modules/flight/flight.controller.ts:224]: Up to you.
I reckon its cheaper to check age vs search cache.
Every user that has !adsRequirements will search cache.

Comment (kksteini) [reply apps/air-discount-scheme/backend/src/app/modules/flight/flight.controller.ts:224]: Ah, that however is a good point

Comment (kksteini) [reply apps/air-discount-scheme/backend/src/app/modules/flight/flight.controller.ts]: Fixed

**Pull request merged**